### PR TITLE
add nameContains option to faFilter

### PR DESCRIPTION
**Pull request by RavenYL**
# Description
add nameContains option to faFilter

### Testing
added test

### Checklist before requesting a review

- [x] I performed a self-review of my code
- [x] If it this a core feature, I added thorough tests
- [x] The command `go fmt` or `make clean` was used on all files included

### CODE REVIEW

File: cmd/faFilter/faFilter.go [hunk 1]
```diff
@@ -66,6 +70,7 @@ func main() {
 	var end *int = flag.Int("end", -1, "Retains the sequence before this position.")
 	var name *string = flag.String("name", "", "Specifies the fasta record name.")
 	var notName *string = flag.String("notName", "", "Returns all fasta records except for this input.")
+	var nameContains *string = flag.String("nameContains", "", "Returns all fasta records whose name contains this input.")
```
Comment (sweavs111) [inline cmd/faFilter/faFilter.go:73]: maybe include in the usage if the string matching has to be case sensitive or not?

**Review by sweavs111 — COMMENTED** [review]
looks good, just one comment about the usage statement

File: cmd/faFilter/faFilter.go (hunk 1 above)
Comment (RavenYL) [reply cmd/faFilter/faFilter.go]: Just updated! Thanks!

**Review by sweavs111 — APPROVED** [review]
looks great!

**Review by rimangan — APPROVED** [review]

**Pull request merged**